py-polars/tests/unit/namespaces/test_string.py

```diff
@@ -793,3 +793,21 @@ def test_string_replace_with_nulls_10124() -> None:
         "n_1": ["O", "O", "O", None, "O", "O", "O", "O"],
         "n_3": ["O", "O", "O", None, "O", "O", "O", "O"],
     }
+
+
+def test_string_extract_groups_lazy_schema_10305() -> None:
```

Comment:
We can make the test more lean by just testing the schema after the expression is selected.

Reply:
Ah, okay. I've changed it to:
```python
    dtype = pl.Struct([pl.Field('candidate', pl.Utf8), pl.Field('ref', pl.Utf8)])
    assert (
        df.select(captures=pl.col("url").str.extract_groups(pattern)).schema
        == {'captures': dtype}
    )
```